orderer/consensus/hlmirbft/chain.go

```diff
@@ -534,23 +554,50 @@ func (c *Chain) Submit(req *orderer.SubmitRequest, sender uint64) error {
 		c.Metrics.ProposalFailures.Add(1)
 		return err
 	}
-
-	leadC := make(chan uint64, 1)
+	submitPayload := req.Payload.GetPayload()
+	isfowarded := checkForwardFlag(submitPayload)
+	if isfowarded {
+		req.Payload.Payload = submitPayload[8:]
```

Comment:
First index should be 9 as the / shouldn't be included in the actual payload. See comment in checkForwardFlag for explanation.